From dc0c6cf4cc670001bb3fdf1ee10474e2533e4b9a Mon Sep 17 00:00:00 2001
From: Ramez Medhat <iramezdev@gmail.com>
Date: Sat, 14 Mar 2026 15:50:46 +0200
Subject: [PATCH] display-arrangement: Update cursor styles for hover and drag
 interactions

The previously used "fleur" cursor type is invalid,
causing a fallback to the default cursor, which in some cases
doesn't match the currently active cursor and leads to inconsistency.

This is fixed by using the "grab" cursor on hover and
"grabbing" cursor on drag.

Fixes #3653
---
 panels/display/cc-display-arrangement.c | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/panels/display/cc-display-arrangement.c b/panels/display/cc-display-arrangement.c
index a55aa6a21a..dd53bcd779 100644
--- a/panels/display/cc-display-arrangement.c
+++ b/panels/display/cc-display-arrangement.c
@@ -670,6 +670,7 @@ on_click_gesture_pressed_cb (CcDisplayArrangement *self,
       self->drag_active = TRUE;
       self->drag_anchor_x = event_x - mon_x;
       self->drag_anchor_y = event_y - mon_y;
+      gtk_widget_set_cursor_from_name (GTK_WIDGET (self), "grabbing");
     }
 
   return TRUE;
@@ -693,7 +694,7 @@ on_click_gesture_released_cb (CcDisplayArrangement *self,
 
   output = cc_display_arrangement_find_monitor_at (self, x, y);
   gtk_widget_set_cursor_from_name (GTK_WIDGET (self),
-                                   output != NULL ? "fleur" : NULL);
+                                   output != NULL ? "grab" : NULL);
 
   /* And queue a redraw to recenter everything */
   gtk_widget_queue_draw (GTK_WIDGET (self));
@@ -724,7 +725,7 @@ on_motion_controller_motion_cb (CcDisplayArrangement *self,
       output = cc_display_arrangement_find_monitor_at (self, x, y);
 
       gtk_widget_set_cursor_from_name (GTK_WIDGET (self),
-                                       output != NULL ? "fleur" : NULL);
+                                       output != NULL ? "grab" : NULL);
       if (self->prelit_output != output)
         gtk_widget_queue_draw (GTK_WIDGET (self));
 
-- 
GitLab

